qiskit/synthesis/discrete_basis/gate_sequence.py

```diff
@@ -106,6 +109,27 @@ def to_circuit(self):
 
         return circuit
 
+    def _to_u2(self):
+        """Creates the U2 matrix corresponding to the stored sequence of gates
+        and the global phase.
+        """
+        u2 = np.eye(2, dtype=complex)
+        for mat in self.gates:
+            u2 = mat.to_matrix().dot(u2)
+        u2 = np.exp(1j * self.global_phase) * u2
+        return u2
+
+    def _to_circuit(self, adjust_phase: bool):
```

Comment:
Could we just move the phase fixing inside this function (and the `_to_dag`)? Or do you think there's a scenario where one would not like to adjust the phase?

Reply:
That's fair. In [683f127](https://github.com/Qiskit/qiskit/pull/14217/commits/683f127592208cbc220ea86d302c778b80e34a8d) I have improved the implementation to do all the phase fixing in `SolovayKitaevDecomposition.run`, including updating the SO(3)-phase to the correct SU(2)-phase, and adjusting that to the original U(2)-phase.